framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react'
+import { CreatableSelect } from 'chakra-react-select'
+import type { Option } from '@northlight/ui'
+import { PlusSolid } from '@northlight/icons'
+import { Icon } from '@chakra-ui/react'
+import type { CreatableSelectDropdownProps } from './types'
+import { customComponents } from './custom-components'
+
+/**
+ * A dropdown component that allows users to select a value from given options
+ * or create a new one if it doesn't exist.
+ * @see {@link https://northlight.dev/reference/creatable-select-dropdown}
+ *
+ * @example (Example)
+ * ##Select the best artist
+ * (?
+ * () => {
+ *   const [artist, setArtist] = useState(null);
+ *   const options = [
+ *     { label: 'Scooter', value: 'scooter' },
+ *     { label: 'Snoop Doggy Dogg', value: 'snoop-dogg' },
+ *   ];
+ *
+ *   const handleOptionChange = (newOption) => {
+ *     setArtist(newOption);
+ *     // Perform additional actions with the selected or created option
+ *   };
+ *
+ *   return (
+ *     <Box align="left">
+ *       <CreatableSelectDropdown
+ *         standardOptions={options}
+ *         onOptionChange={handleOptionChange}
+ *         width="300px"
+ *       />
+ *       { artist && artist.value !== 'custom-option' && (
+ *         <H3 py={8}>The best artist is: { artist.label }</H3>
+ *       ) }
+ *     </Box>
+ *   );
+ * }
+ * ?)
+ */
+
+export const CreatableSelectDropdown = ({
+  standardOptions,
+  initialPlaceholder = 'Select or create...',
+  addOptionPlaceholder = 'Enter text...',
+  creationOption = {
+    label: 'Add option...',
+    value: 'custom-option',
+  },
+  onOptionChange,
+  width = '100%',
+  variant = 'outline',
+}: CreatableSelectDropdownProps) => {
+  const [ options, setOptions ] = useState(standardOptions)
+  const [ selectedOption, setSelectedOption ] = useState<Option | null>(null)
+  const [ newOptionText, setNewOptionText ] = useState('')
+  const [ newOptionPlaceholder, setNewOptionPlaceholder ] = useState(initialPlaceholder)
+  const [ menuIsOpen, setMenuIsOpen ] = useState(false)
+
+  const handleInputChange = (newValue: string) => {
+    setNewOptionText(newValue)
+  }
+
+  const handleCreateOption = () => {
+    const newOptionTextLower = newOptionText.toLowerCase()
+    const newOptionExists = options.some((option) =>
+      option.value.toLowerCase() === newOptionTextLower)
+
+    if (!newOptionExists) {
+      const newOption = { label: newOptionText, value: newOptionText }
+      setOptions((currentOptions) => [ ...currentOptions, newOption ])
+      setSelectedOption(newOption)
+      onOptionChange(newOption)
+    }
+
+    setNewOptionText('')
+    setMenuIsOpen(false)
+    setNewOptionPlaceholder(initialPlaceholder)
+  }
+
+  const handleChange = (option: Option) => {
+    if (option?.value === 'custom-option') {
```

Comment:
Looks unconventional to use `custom-option` here. I will leave it to @SebCodesTheWeb :)

Reply:
Hmmm yes, this will techincally lead to a bug where the user cannot create an option called 'custom-option'. If I was going this route with I would probably have some random id instead. More preferably however would be if you added some isCreation property to the option. You would have to extend the type of Option if you do this.
![Screenshot 1445-07-27 at 16 04 57](https://github.com/mediatool/northlight/assets/90923099/ea3a1c81-9992-4662-8c2a-8c0ae3e4ddb3)

Reply:
done! went with the extend solution.

Reply:
Yes. You would also have to do (option?.isCreation ) instead of (option?.value === 'custom-value')